src/Datepicker/Datepicker.tsx

```diff
@@ -84,44 +74,14 @@ export const Datepicker: FC<DatepickerProps> = ({
         active: activeDay ? isSameDay(date, activeDay) : false,
         disabled:
           (!isToday(date) &&
-            !isWithinInterval(date, { start: startDay, end: endDay })) ||
+            !isWithinInterval(date, { start: startDate, end: endDate })) ||
           (disableWeekend && isWeekend(date)),
       })
     }
 
-    return {
-      monthName,
-      filteredDays,
-    }
+    return filteredDays
   }
 
-  const combineAvailableDays = () => {
-    const today = fromDate
-    const endDay = addDays(today, range)
-    const sameMonth = isSameMonth(today, endDay)
-
-    let availableDays = []
-
-    // Check if we should generate dates from current for next month
-    if (!sameMonth) {
-      const thisMonth = generateDaysForMonth(today, endDay)
-      const nextMonth = generateDaysForMonth(
-        startOfMonth(addMonths(today, 1)),
-        endDay,
-      )
-
-      availableDays = [thisMonth, nextMonth]
-    } else {
-      availableDays = [generateDaysForMonth(today, endDay)]
-    }
-
-    setListDays(availableDays)
-  }
```

Comment:
We were calculating all month before, I switched to only generate the days of the current displayed month